lib/ex_plasma/output/type/abstract_payment.ex

```diff
@@ -0,0 +1,107 @@
+defmodule ExPlasma.Output.Type.AbstractPayment do
```

Comment:
another approach would be to use a macro that would inject overridable functions

Reply:
also, what do you think about defining a behaviour that all modules including `AbstractPayment` would implement?

Reply:
I would vote for a behaviour if any, but the question is if it's worth the abstraction since we have two use cases. is it @mederic-p ? up to you

Reply:
I like having a behaviour here, will add